### DISCUSSION TITLE
Wish for signerOrProvider in hooks

### DISCUSSION THREAD

**Discussion by unsphere** [wevm/wagmi#809] [Ideas]
In a lot of cases, I need to call smart contract views to show data without having an actual wallet connection or to show a specific balance from another chain than the one that is connected. So at the moment, I use my own useContract hook and a dummy wallet private key with ethers to call the smart contract functions each n second. I want to fully stick to wagmi, so it would be great to have signerOrProvider in more of your hooks like useContractRead and useContractReads than just the useContract hook.

**Comment by tmm** [comment]
`useContractRead` and `useContractReads` work without a wallet connection. Have you tried passing in [`chainId`](https://wagmi.sh/docs/hooks/useContractRead)?

**Reply by unsphere** [reply]
Ah thanks it can be so easy :) Never checked it using chainId...

It is working however when I use watch I noticed that it will refresh based on the new blocks of eth although I use chainId = 56 which produces blocks about every 3 secs, so I need a faster polling.

You can also see it in the useBlockNumber hook that something is not right, because it shows the correct blockNumber on first render for chainId 56, and on next render it shows the eth blockNumber. 

I prepared a sandbox: [https://codesandbox.io/s/wagmi-hooks-test-lxm8rw?file=/src/App.js](https://codesandbox.io/s/wagmi-hooks-test-lxm8rw?file=/src/App.js)

Do I have a wrong configuration? Thanks for your help!
